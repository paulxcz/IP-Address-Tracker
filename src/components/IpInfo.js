import { faSpinner } from "@fortawesome/free-solid-svg-icons";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import React, { useEffect, useState } from "react";

import "../../src/styles/IpInfor/IpInfo.css";
import { getIpInfo } from "../helpers/getIpInfo";

export const IpInfo = ({ ipToFind, setcords }) => {
  
  const [IpInformation, setIpInformation] = useState({
    ip: "",
    location: {},
    isp: "",
  });

  useEffect(() => {
    getIpInfo(ipToFind)
      .then(({ ip, location, isp }) =>
        setIpInformation({
          ip,
          location,
          isp,
        })
      )
      .catch((err) => {
        console.error("ERROR: ", err.message);
        setIpInformation({
          ip: "-",
          location: {
            city: "-",
            country: "-",
            timezone: "-",
          },
          isp: "-",
        });
        alert("IP INVALID");
      });
  }, [ipToFind]);

  const { ip, location, isp } = IpInformation;
  const { city, lat, lng, country, timezone } = location;
    
  useEffect(() => {
    (lat !== undefined &&  lng !== undefined) && setcords({
      lat,
      lng
    });
  }, [lat, lng])
  
  // console.log(lat, lng);

  return (
    <>
      <div className="ip-info-container" >
        <div className="info">
          <span>IP ADRESS</span>
          {ip === "" ? <FontAwesomeIcon icon={faSpinner} spin /> : <b>{ip}</b>}
        </div>

        <div className="info">
          <span>LOCATION</span>
          {(city && country) === undefined ? (
            <FontAwesomeIcon icon={faSpinner} spin />
          ) : (
            <b>{`${city}, ${country}`}</b>
          )}
        </div>

        <div className="info">
          <span>TIMEZONE</span>
          {timezone === undefined ? (
            <FontAwesomeIcon icon={faSpinner} spin />
          ) : (
            <b>{timezone}</b>
          )}
        </div>

        <div className="info" id = "info-isp">
          <span>ISP</span>
          {isp === "" ? (
            <FontAwesomeIcon icon={faSpinner} spin />
          ) : (
            <b>{isp}</b>
          )}
        </div>
      </div>
    </>
  );
};

