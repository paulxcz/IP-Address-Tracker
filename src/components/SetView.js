import { useMap } from "react-leaflet";

export const SetView = ({position}) => {
  const map = useMap()
  map.setView(position);
  return null;
};
