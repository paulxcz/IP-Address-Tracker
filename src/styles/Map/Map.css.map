{
    "version": 3,
    "mappings": "AAAA,AAAA,MAAM,CAAC;EACL,MAAM,EAAE,KAAK;EACb,KAAK,EAAE,IAAI;EACX,OAAO,EAAE,CAAC;CACX;;AAED,MAAM,CAAC,MAAM,MAAM,SAAS,EAAE,MAAM;EAClC,AAAA,MAAM,CAAA;IACJ,MAAM,EAAE,KAAK;GACd",
    "sources": [
        "Map.scss"
    ],
    "names": [],
    "file": "Map.css"
}