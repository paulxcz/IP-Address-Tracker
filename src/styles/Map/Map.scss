#mapid {
  height: 451px;
  width: 100%;
  z-index: 0;
}

@media screen and(min-width: 1024px) {
  #mapid{
    height: 520px;
  }
}