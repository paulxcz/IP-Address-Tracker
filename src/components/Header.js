import React, { useState } from "react";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faChevronRight } from "@fortawesome/free-solid-svg-icons";
import { IpInfo } from "./IpInfo";


import "../styles/Header/Header.css";
export default function Header({setcords}) {

  const [ipToFind, setipToFind] = useState({
    visibility : false,
    ip : "",
  });

  const { visibility, ip } = ipToFind;

  
  return (
    <>
      <header>
        <h1>IP Address Tracker</h1>
        <div className="search-box">
          <input type="text"
            value = {ip}
            onChange = { ({target}) => { setipToFind({
              visibility: false,
              ip: target.value
            }) }}
            maxLength="15"
            placeholder = "Search for any IP adress or domain"
            />
          <button
            onClick={(e) => {
              e.preventDefault();
              setipToFind({
                ...ipToFind,
                visibility: !visibility,
              });
            }}
          >
            <FontAwesomeIcon icon={faChevronRight} className="right-arrow" />
          </button>
        </div>

        {visibility && <IpInfo ipToFind={ip} setcords={setcords}/>}
      </header>
    </>
  );
}
