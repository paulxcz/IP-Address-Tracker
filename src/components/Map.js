import React from "react";
import "../../src/styles/Map/Map.css";
import { MapContainer, TileLayer, Marker, Popup } from "react-leaflet";
import { SetView } from "./SetView";
import * as L from "leaflet";

export const Map = ({ cords }) => {
  const blackIcon = new L.Icon({
    iconUrl: 'https://raw.githubusercontent.com/pointhi/leaflet-color-markers/master/img/marker-icon-2x-black.png',
    shadowUrl: 'https://cdnjs.cloudflare.com/ajax/libs/leaflet/0.7.7/images/marker-shadow.png',
    iconSize: [25, 41],
    iconAnchor: [12, 41],
    popupAnchor: [1, -34],
    shadowSize: [41, 41]
  });
  return (
    <>
      <MapContainer
        id="mapid"
        center={[0, 0]}
        zoom={14}
        scrollWheelZoom={false}
        touchZoom = "center"
        tap = "true"
      >
        <TileLayer
          attribution='&copy; <a href="http://osm.org/copyright">OpenStreetMap</a> contributors'
          url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
        />
        <SetView position={[cords.lat, cords.lng]} />
        <Marker position={[cords.lat, cords.lng]} icon={blackIcon}>
          <Popup>
            Ip location.
          </Popup>
        </Marker>
      </MapContainer>
    </>
  );
};
