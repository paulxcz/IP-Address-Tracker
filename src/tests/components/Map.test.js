import '@testing-library/jest-dom';
import {shallow} from 'enzyme'
import {Map} from '../../components/Map'

describe('Tests on <Map />', () => {
    const cords = {
        lat: 35.63304462061077,
        lng: 139.8803621105462
    }

    

    test('should show <Map /> correctly', () => {
        const wraper = shallow(<Map cords = {cords} />)
        expect( wraper ).toMatchSnapshot();

    })
    

})
