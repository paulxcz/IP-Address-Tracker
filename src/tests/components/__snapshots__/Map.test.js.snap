// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <Map /> should show <Map /> correctly 1`] = `
<Fragment>
  <MapContainer
    center={
      Array [
        0,
        0,
      ]
    }
    id="mapid"
    scrollWheelZoom={false}
    zoom={14}
  >
    <ForwardRef(LeafComponent)
      attribution="Â© <a href=\\"http://osm.org/copyright\\">OpenStreetMap</a> contributors"
      url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
    />
    <SetView
      position={
        Array [
          35.63304462061077,
          139.8803621105462,
        ]
      }
    />
    <ForwardRef(ContainerComponent)
      icon={
        NewClass {
          "_initHooksCalled": true,
          "options": Object {
            "iconAnchor": Array [
              12,
              41,
            ],
            "iconSize": Array [
              25,
              41,
            ],
            "iconUrl": "https://raw.githubusercontent.com/pointhi/leaflet-color-markers/master/img/marker-icon-2x-black.png",
            "popupAnchor": Array [
              1,
              -34,
            ],
            "shadowSize": Array [
              41,
              41,
            ],
            "shadowUrl": "https://cdnjs.cloudflare.com/ajax/libs/leaflet/0.7.7/images/marker-shadow.png",
          },
        }
      }
      position={
        Array [
          35.63304462061077,
          139.8803621105462,
        ]
      }
    >
      <ForwardRef(OverlayComponent)>
        A pretty CSS3 popup. 
        <br />
         Easily customizable.
      </ForwardRef(OverlayComponent)>
    </ForwardRef(ContainerComponent)>
  </MapContainer>
</Fragment>
`;
